tapchannel/aux_closer.go

```diff
@@ -110,13 +123,14 @@ func createCloseAlloc(isLocal, isInitiator bool, closeAsset *asset.Asset,
 
 			return CommitAllocationToRemote
 		}(),
-		SplitRoot:           isInitiator,
-		InternalKey:         shutdownMsg.AssetInternalKey.Val,
-		ScriptKey:           asset.NewScriptKey(&scriptKey),
-		Amount:              closeAsset.Amount,
-		AssetVersion:        asset.V0,
-		BtcAmount:           tapsend.DummyAmtSats,
-		SortTaprootKeyBytes: sortKeyBytes,
+		SplitRoot:            isInitiator,
+		InternalKey:          shutdownMsg.AssetInternalKey.Val,
+		ScriptKey:            asset.NewScriptKey(&scriptKey),
+		Amount:               closeAsset.Amount,
+		AssetVersion:         asset.V0,
+		BtcAmount:            tapsend.DummyAmtSats,
+		SortTaprootKeyBytes:  sortKeyBytes,
+		ProofDeliveryAddress: proofDeliveryUrl,
```

Comment:
Ah existing tests assumed that they were all using the same courier.

Reply:
Before this commit the respondent to the shutdown process didn't actually know what proof courier the final proof would be uploaded to. Now the desired delivery address is sent along with the internal and script key, so the non-shutdown-initiator will know where to look for the proof.